Cargo update breaks build

I think this is the same issues that was addressed in 

- https://github.com/SpectrumXYZ/jellyfish/pull/62

<details>
<summary>  Cargo update output </summary>

```
 lulu  bix  ~  r t cape  $  cargo update                                                                                                                                                 0.018s  main  ❄  10:29:25 
    Updating crates.io index
    Updating git repository `https://github.com/arkworks-rs/poly-commit/`
    Updating git repository `https://github.com/gakonst/ethers-rs`
    Updating git repository `ssh://git@github.com/SpectrumXYZ/jellyfish.git`
    Updating git repository `ssh://git@github.com/SpectrumXYZ/spectrum.git`
    Updating git repository `https://github.com/gakonst/ethers-rs`
    Updating git repository `ssh://git@github.com/SpectrumXYZ/jellyfish.git`
    Updating git repository `ssh://git@github.com/SpectrumXYZ/phaselock.git`
    Updating git repository `ssh://git@github.com/SpectrumXYZ/tagged-base64.git`
    Updating git repository `https://github.com/arkworks-rs/curves`
    Updating `https://dl.cloudsmith.io/basic/translucence/tl/cargo/index.git` index
    Updating git repository `ssh://git@github.com/SpectrumXYZ/phaselock.git`
    Updating git repository `https://github.com/arkworks-rs/curves`
    Updating git repository `https://github.com/rust-ethereum/ethabi`
    Updating git repository `https://github.com/hyperledger-labs/solang`
    Updating anyhow v1.0.51 -> v1.0.52
    Updating ark-bls12-377 v0.3.0 (https://github.com/arkworks-rs/curves#677b4ae7) -> #d1ced15c
    Updating ark-bw6-761 v0.3.0 (https://github.com/arkworks-rs/curves#677b4ae7) -> #d1ced15c
    Updating ark-ed-on-bls12-377 v0.3.0 (https://github.com/arkworks-rs/curves#677b4ae7) -> #d1ced15c
    Updating ark-ed-on-bls12-381-bandersnatch v0.3.0 (https://github.com/arkworks-rs/curves#677b4ae7) -> #d1ced15c
      Adding ascii-canvas v3.0.0
    Updating base64ct v1.3.0 -> v1.3.3
    Updating blake3 v1.2.0 -> v1.3.0
    Updating bumpalo v3.8.0 -> v3.9.1
    Updating cache-padded v1.1.1 -> v1.2.0
    Updating clipboard-win v4.2.2 -> v4.3.0
    Updating coins-bip32 v0.3.0 -> v0.6.0
    Updating coins-bip39 v0.3.0 -> v0.6.0
    Removing const-oid v0.6.2
    Updating const_fn v0.4.8 -> v0.4.9
    Updating convert_case v0.4.0 -> v0.5.0
    Removing core-foundation v0.9.2
    Removing core-foundation-sys v0.8.3
    Updating crossbeam-channel v0.5.1 -> v0.5.2
    Updating crossbeam-epoch v0.9.5 -> v0.9.6
    Updating crossbeam-utils v0.8.5 -> v0.8.6
    Removing crypto-bigint v0.2.11
    Updating curl v0.4.41 -> v0.4.42
    Updating curl-sys v0.4.51+curl-7.80.0 -> v0.4.52+curl-7.81.0
    Removing der v0.4.5
      Adding diff v0.1.12
      Adding dunce v1.0.2
    Updating ecdsa v0.12.4 -> v0.13.4
    Removing elliptic-curve v0.10.6
    Removing elliptic-curve v0.11.5
      Adding elliptic-curve v0.11.6
      Adding ena v0.14.0
    Removing ethabi v15.0.0
      Adding ethabi v16.0.0 (https://github.com/rust-ethereum/ethabi?branch=master#7f4bb3d6)
    Updating ethers v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
      Adding ethers-addressbook v0.1.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
      Adding ethers-addressbook v0.1.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Updating ethers-contract v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-contract v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-contract-abigen v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-contract-abigen v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-contract-derive v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-contract-derive v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-core v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-core v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-etherscan v0.2.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-etherscan v0.2.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-middleware v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-middleware v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-providers v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-providers v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-signers v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-signers v0.6.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating ethers-solc v0.1.0 (https://github.com/gakonst/ethers-rs?branch=master#5bf3f735) -> #3b0b3132
    Updating ethers-solc v0.1.0 (https://github.com/gakonst/ethers-rs#5bf3f735) -> #3b0b3132
    Updating fastrand v1.5.0 -> v1.6.0
    Updating ff v0.10.1 -> v0.11.0
      Adding fixedbitset v0.2.0
    Removing foreign-types v0.3.2
    Removing foreign-types-shared v0.1.1
    Updating futures v0.3.17 -> v0.3.19
    Updating futures-channel v0.3.17 -> v0.3.19
    Updating futures-core v0.3.17 -> v0.3.19
    Updating futures-executor v0.3.17 -> v0.3.19
    Updating futures-io v0.3.17 -> v0.3.19
    Updating futures-macro v0.3.17 -> v0.3.19
    Updating futures-sink v0.3.17 -> v0.3.19
    Updating futures-task v0.3.17 -> v0.3.19
    Updating futures-util v0.3.17 -> v0.3.19
    Updating generic-array v0.14.4 -> v0.14.5
    Updating group v0.10.0 -> v0.11.0
    Updating h2 v0.3.9 -> v0.3.10
    Updating http v0.2.5 -> v0.2.6
    Removing hyper-tls v0.5.0
    Updating indexmap v1.7.0 -> v1.8.0
    Updating jf-plonk v0.0.1 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?branch=main#af112cf1) -> #d665de4b
    Updating jf-primitives v0.0.1 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?branch=main#af112cf1) -> #d665de4b
    Updating jf-rescue v0.0.1 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?branch=main#af112cf1) -> #d665de4b
    Updating jf-utils v0.0.1 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?branch=main#af112cf1) -> #d665de4b
    Updating jf-utils-derive v0.1.0 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?branch=main#af112cf1) -> #d665de4b
    Updating k256 v0.9.6 -> v0.10.1
      Adding lalrpop v0.19.6
      Adding lalrpop-util v0.19.6
    Removing native-tls v0.2.8
      Adding new_debug_unreachable v1.0.4
    Updating nix v0.23.0 -> v0.23.1
      Adding num-rational v0.4.0
    Updating num_cpus v1.13.0 -> v1.13.1
    Removing openssl v0.10.38
      Adding petgraph v0.5.1
      Adding phf v0.10.1
      Adding phf_generator v0.10.0
      Adding phf_macros v0.10.0
      Adding phf_shared v0.8.0
      Adding phf_shared v0.10.0
      Adding pico-args v0.4.2
    Updating pin-project v1.0.8 -> v1.0.10
    Updating pin-project-internal v1.0.8 -> v1.0.10
    Updating pin-project-lite v0.2.7 -> v0.2.8
    Updating pkcs8 v0.7.6 -> v0.8.0
    Updating ppv-lite86 v0.2.15 -> v0.2.16
      Adding precomputed-hash v0.1.1
    Removing proc-macro-nested v0.1.7
    Updating proc-macro2 v1.0.34 -> v1.0.36
    Updating quote v1.0.10 -> v1.0.14
    Updating reqwest v0.11.7 -> v0.11.8
      Adding rfc6979 v0.1.0
    Updating rustyline v9.1.1 -> v9.1.2
      Adding sec1 v0.2.1
    Removing security-framework v2.4.2
    Removing security-framework-sys v2.4.2
    Updating serde v1.0.131 -> v1.0.133
    Updating serde_derive v1.0.131 -> v1.0.133
    Updating serde_json v1.0.73 -> v1.0.74
    Updating sha2 v0.9.8 -> v0.9.9
    Removing sha2-asm v0.6.2
    Updating signal-hook v0.3.12 -> v0.3.13
    Updating signature v1.3.2 -> v1.4.0
      Adding siphasher v0.3.7
      Adding solang-parser v0.1.1 (https://github.com/hyperledger-labs/solang#190f7205)
    Updating spki v0.4.1 -> v0.5.4
      Adding string_cache v0.8.2
    Updating syn v1.0.82 -> v1.0.85
    Updating tagged-base64 v0.1.0 (ssh://git@github.com/SpectrumXYZ/tagged-base64.git?branch=main#02b302b5) -> #408f79ee
    Updating tempfile v3.2.0 -> v3.3.0
      Adding term v0.7.0
    Removing tokio-native-tls v0.3.0
    Updating tokio-rustls v0.23.1 -> v0.23.2
    Updating tracing-subscriber v0.3.3 -> v0.3.5
    Updating typenum v1.14.0 -> v1.15.0
    Updating version_check v0.9.3 -> v0.9.4
    Removing webpki v0.21.4
    Updating webpki-roots v0.21.1 -> v0.22.2
 lulu  bix  ~  r t cape  $  cargo check                                                                                                                                                 25.951s  main  ❄  10:34:13 
    Checking ansi_term v0.12.1
    Checking siphasher v0.3.7
    Checking paw-raw v1.0.0
    Checking vec_map v0.8.2
    Checking strsim v0.8.0
    Checking pipeline v0.5.0
    Checking itertools v0.9.0
    Checking regex-automata v0.1.10
    Checking crossbeam-utils v0.8.6
    Checking proc-macro2 v1.0.36
    Checking spki v0.5.4
    Checking num-bigint v0.4.3
    Checking curl-sys v0.4.52+curl-7.81.0
    Checking lalrpop-util v0.19.6
    Checking serde v1.0.133
    Checking value-bag v1.0.0-alpha.8
    Checking shellexpand v2.1.0
    Checking thiserror v1.0.30
    Checking futures-util v0.3.19
    Checking pin-project v1.0.10
    Checking indexmap v1.8.0
    Checking time-macros v0.1.1
    Checking snafu v0.6.10
    Checking arbitrary v1.0.3
    Checking zeroize v1.3.0
    Checking failure v0.1.8
    Checking ff v0.6.0
   Compiling hyper-rustls v0.23.0
    Checking markdown v0.3.0
    Checking paw v1.0.0
    Checking flume v0.10.9
    Checking time v0.2.27
    Checking group v0.6.0
    Checking phf_shared v0.10.0
    Checking clap v2.34.0
    Checking curl v0.4.42
    Checking pkcs8 v0.8.0
    Checking merlin v3.0.0
    Checking crossbeam-epoch v0.9.6
    Checking crossbeam-channel v0.5.2
    Checking rust-argon2 v0.8.3
    Checking log v0.4.14
   Compiling reqwest v0.11.8
    Checking phf v0.10.1
    Checking quote v1.0.14
    Checking pairing v0.16.0
    Checking crossbeam-deque v0.8.1
    Checking syn v1.0.85
    Checking tracing v0.1.29
    Checking polling v2.2.0
    Checking mio v0.7.14
    Checking kv-log-macro v1.0.7
    Checking rustls v0.20.2
    Checking want v0.3.0
    Checking tracing-log v0.1.2
    Checking rustyline v9.1.2
    Checking rayon-core v1.9.1
    Checking async-io v1.6.0
    Checking matchers v0.0.1
    Checking tracing-futures v0.2.5
    Checking tracing-error v0.2.0
    Checking tracing-unwrap v0.9.2
    Checking rayon v1.5.1
    Checking async-process v1.3.0
    Checking async-global-executor v2.0.2
    Checking async-std v1.10.0
    Checking num-rational v0.4.0
    Checking solang-parser v0.1.1 (https://github.com/hyperledger-labs/solang#190f7205)
   Compiling ethers-contract-abigen v0.6.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
   Compiling ethers-contract-abigen v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking structopt v0.3.25
    Checking futures-executor v0.3.19
    Checking futures v0.3.19
    Checking generic-array v0.14.5
    Checking rand_chacha v0.3.1
    Checking serde_json v1.0.74
    Checking bytes v1.1.0
    Checking url v2.2.2
    Checking impl-serde v0.3.2
    Checking parity-scale-codec v2.3.1
    Checking serde_urlencoded v0.7.0
    Checking cargo-platform v0.1.2
    Checking camino v1.0.5
    Checking semver v1.0.4
    Checking bincode v1.3.3
    Checking chrono v0.4.19
    Checking serde_qs v0.8.5
    Checking uuid v0.8.2
    Checking threshold_crypto v0.4.0
    Checking bstr v0.2.17
    Checking tracing-serde v0.1.2
    Checking toml v0.5.8
    Checking serde_with v1.11.0
    Checking serde_cbor v0.11.2
    Checking rand v0.8.4
    Checking http v0.2.6
    Checking rlp v0.5.1
    Checking tokio v1.15.0
    Checking input_buffer v0.4.0
    Checking digest v0.9.0
    Checking block-buffer v0.9.0
    Checking crypto-mac v0.11.1
    Checking cipher v0.3.0
    Checking crypto-bigint v0.3.2
    Checking sec1 v0.2.1
    Checking universal-hash v0.4.1
    Checking crypto-mac v0.8.0
    Checking cipher v0.2.5
    Checking aead v0.4.3
    Checking crypto-mac v0.10.1
    Checking block-buffer v0.10.0
    Checking crypto-common v0.1.1
    Checking aead v0.3.2
    Checking hmac v0.11.0
    Checking signature v1.4.0
    Checking curve25519-dalek v3.2.0
    Checking sha2 v0.9.9
    Checking sha3 v0.9.1
    Checking sha-1 v0.9.8
    Checking ripemd160 v0.9.1
    Checking salsa20 v0.8.1
    Checking chacha20 v0.7.3
    Checking aes v0.7.5
    Checking ctr v0.7.0
    Checking chacha20 v0.8.1
    Checking csv v1.1.6
    Checking blake2 v0.9.2
    Checking blake3 v0.3.8
    Checking hmac v0.8.1
    Checking poly1305 v0.7.2
    Checking polyval v0.4.5
    Checking impl-rlp v0.3.0
    Checking hmac v0.10.1
    Checking aes-soft v0.6.4
    Checking ctr v0.6.0
    Checking digest v0.10.1
    Checking impl-codec v0.5.1
    Checking cargo_metadata v0.14.1
    Checking wasm-bindgen v0.2.78
    Checking serde-aux v3.0.1
    Checking femme v2.1.1
    Checking hkdf v0.10.0
    Checking pbkdf2 v0.8.0
    Checking ghash v0.3.1
    Checking procfs v0.11.1
    Checking tracing-subscriber v0.2.25
    Checking xsalsa20poly1305 v0.7.2
    Checking chacha20poly1305 v0.8.2
    Checking blake3 v1.3.0
    Checking md-5 v0.10.0
    Checking ark-std v0.3.0
    Checking fixed-hash v0.7.0
    Checking aes v0.6.0
    Checking coins-core v0.2.2
    Checking ethbloom v0.11.1
    Checking primitive-types v0.10.1
    Checking scrypt v0.7.0
    Checking aes-gcm v0.8.0
    Checking ark-serialize v0.3.0
    Checking rfc6979 v0.1.0
    Checking elliptic-curve v0.11.6
    Checking cookie v0.14.4
    Checking eth-keystore v0.3.0
    Checking x25519-dalek v1.2.0
    Checking ark-ff v0.3.0
    Checking atomic_store v0.0.2 (registry `https://dl.cloudsmith.io/basic/translucence/tl/cargo/index.git`)
    Checking crypto_box v0.6.1
    Checking ecdsa v0.13.4
   Compiling ethers-contract-derive v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
   Compiling ethers-contract-derive v0.6.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking js-sys v0.3.55
    Checking console_error_panic_hook v0.1.7
    Checking k256 v0.10.1
    Checking coins-bip32 v0.6.0
    Checking http-body v0.4.4
    Checking isahc v0.9.14
    Checking tungstenite v0.16.0
    Checking tungstenite v0.15.0
    Checking tungstenite v0.13.0
    Checking ethereum-types v0.12.1
    Checking tracing-distributed v0.3.1
    Checking coins-bip39 v0.6.0
    Checking http-types v2.12.0
    Checking async-session v2.0.1
    Checking async-scoped v0.7.0
    Checking async_executors v0.4.2
    Checking async-tungstenite v0.16.1
    Checking async-tungstenite v0.13.1
    Checking async-tungstenite v0.15.0
    Checking ethabi v16.0.0 (https://github.com/rust-ethereum/ethabi?branch=master#7f4bb3d6)
    Checking ethers-core v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking ethers-core v0.6.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking tokio-util v0.6.9
    Checking tokio-rustls v0.23.2
    Checking http-client v6.5.1
    Checking async-sse v4.1.0
    Checking async-h1 v2.3.3
    Checking h2 v0.3.10
    Checking web-sys v0.3.55
    Checking wasm-bindgen-futures v0.4.28
    Checking tide v0.16.0
    Checking tide-websockets v0.4.0
    Checking ethers-signers v0.6.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking ethers-addressbook v0.1.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking ethers-solc v0.1.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking ethers-signers v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking ethers-addressbook v0.1.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking ethers-solc v0.1.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking tagged-base64 v0.1.0 (ssh://git@github.com/SpectrumXYZ/tagged-base64.git?branch=main#408f79ee)
    Checking surf v2.3.2
    Checking hyper v0.14.16
    Checking ark-ec v0.3.0
    Checking ark-poly v0.3.0
    Checking ethers-providers v0.6.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking ethers-providers v0.6.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking ethers-etherscan v0.2.0 (https://github.com/gakonst/ethers-rs#3b0b3132)
    Checking ethers-etherscan v0.2.0 (https://github.com/gakonst/ethers-rs?branch=master#3b0b3132)
    Checking ark-bls12-381 v0.3.0
    Checking ark-bn254 v0.3.0
    Checking ark-sponge v0.3.0
    Checking ark-bls12-377 v0.3.0 (https://github.com/arkworks-rs/curves#d1ced15c)
    Checking jf-utils v0.0.1 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?rev=2a40d01c938cdcc716071af5a0dc9b3242181c2c#2a40d01c)
    Checking ark-bls12-377 v0.3.0 (https://github.com/arkworks-rs/curves?rev=677b4ae751a274037880ede86e9b6f30f62635af#677b4ae7)
    Checking jf-utils v0.0.1 (ssh://git@github.com/SpectrumXYZ/jellyfish.git?branch=main#d665de4b)
error[E0438]: const `COFACTOR` is not a member of trait `ModelParameters`
  --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g1.rs:23:5
   |
23 |     const COFACTOR: &'static [u64] = &[0x0, 0x170b5d4430000000];
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `ModelParameters`

error[E0438]: const `COFACTOR_INV` is not a member of trait `ModelParameters`
  --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g1.rs:28:5
   |
28 |     const COFACTOR_INV: Fr = field_new!(Fr, "5285428838741532253824584287042945485047145357130994810877");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `ModelParameters`

error[E0438]: const `COFACTOR` is not a member of trait `ModelParameters`
  --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g2.rs:16:5
   |
16 | /     const COFACTOR: &'static [u64] = &[
17 | |         0x0000000000000001,
18 | |         0x452217cc90000000,
19 | |         0xa0f3622fba094800,
...  |
24 | |         0x26ba558ae9562a,
25 | |     ];
   | |______^ not a member of trait `ModelParameters`

error[E0438]: const `COFACTOR_INV` is not a member of trait `ModelParameters`
  --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g2.rs:30:5
   |
30 |     const COFACTOR_INV: Fr = field_new!(Fr, "6764900296503390671038341982857278410319949526107311149686707033187604810669");
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ not a member of trait `ModelParameters`

error[E0046]: not all trait items implemented, missing: `COFACTOR`, `COFACTOR_INV`
  --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g1.rs:31:1
   |
31 | impl SWModelParameters for Parameters {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `COFACTOR`, `COFACTOR_INV` in implementation
   |
   = help: implement the missing item: `const COFACTOR: &'static [u64] = value;`
   = help: implement the missing item: `const COFACTOR_INV: <Self as ModelParameters>::ScalarField = value;`

error[E0046]: not all trait items implemented, missing: `COFACTOR`, `COFACTOR_INV`
   --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g1.rs:101:1
    |
101 | impl TEModelParameters for Parameters {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `COFACTOR`, `COFACTOR_INV` in implementation
    |
    = help: implement the missing item: `const COFACTOR: &'static [u64] = value;`
    = help: implement the missing item: `const COFACTOR_INV: <Self as ModelParameters>::ScalarField = value;`

error[E0046]: not all trait items implemented, missing: `COFACTOR`, `COFACTOR_INV`
  --> /home/lulu/.cargo/git/checkouts/curves-a0d652819d10cde2/d1ced15/bls12_377/src/curves/g2.rs:33:1
   |
33 | impl SWModelParameters for Parameters {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `COFACTOR`, `COFACTOR_INV` in implementation
   |
   = help: implement the missing item: `const COFACTOR: &'static [u64] = value;`
   = help: implement the missing item: `const COFACTOR_INV: <Self as ModelParameters>::ScalarField = value;`

Some errors have detailed explanations: E0046, E0438.
For more information about an error, try `rustc --explain E0046`.
error: could not compile `ark-bls12-377` due to 7 previous errors
warning: build failed, waiting for other jobs to finish...
error: build failed
```
</details>

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,9 +6,3 @@ members = [
     "relayer",
     "wallet",
 ]
-
-# Commit after the pinned one below breaks ethers-rs compiliation.
-# NOTE: the git url needs to be different to patch, hence the double slash
-# See https://github.com/rust-lang/cargo/issues/5478#issuecomment-522719793
-[patch."https://github.com/hyperledger-labs/solang"]
-solang-parser = { git = "https://github.com/hyperledger-labs//solang", rev = "0ea95870125a7ac16f191e5d14cb5ed9e861b833" }

--- a/flake.nix
+++ b/flake.nix
@@ -70,7 +70,7 @@
             pythonEnv
           ];
 
-          stableToolchain = pkgs.rust-bin.stable."1.56.0".minimal.override {
+          stableToolchain = pkgs.rust-bin.stable."1.56.1".minimal.override {
             extensions = [ "rustfmt" "clippy" "llvm-tools-preview" "rust-src" ];
           };
           rustDeps = with pkgs; [

